### ISSUE TITLE
DEVELOPMENT #3: To be fully vanilla feature and then 1.0 OpenApoc release

### ISSUE THREAD

**Issue by makus82** [OpenApoc/OpenApoc#265]
Then, to be fully vanilla feature complete, we also need:
- [ ] Better battle AI (behavior, taking cover, more intelligent attack patterns)
-- [ ] advanced behavior, 
-- [ ] taking cover, 
-- [ ] more intelligent attack patterns
- [ ] Better city AI (craft retreat when damaged, other stuff)
- [ ] Proper ground vehicles (occupying lanes, blocking, overtaking etc.)
- [ ] Proper location screen (assigning agents to vehicles)
- [ ] Some fixes to Ufopaedia display
- [ ] Proper transferring agents
- [ ] UI Tooltips
- [ ] Controls for editing text (naming soldiers and other property)
- [ ] Agent medals and statistics
- [ ] Agent name generator (more than 10 names and surnames)
- [ ] Colored text support

And then, to be a full 1.0 OpenApoc release, we also need:
- [ ] Different file format than xml for storing save files (at least), so that save/loading takes reasonable amount of time
- [ ] Different handling of game data (separation of "rules" and "gamestate", so that for example you can modify a research project in a mod, add that mod to your playthrough midgame, and not have to lose all progress made on that project, or later remove the project mod, and have the changes reversed but research state persist)
- [ ] Maybe something else that didn't come up to mind immediately
- [ ] Close all issues

**Comment by Istrebitel** [comment]
Really, to have a playable OpenApoc, we need ONE THING: AI

That's it. Road vehicles that don't behave appropriately, illegal flyers not spawning, all that is really insignificant, really. It's fluff, it's mostly just ignored or a nuisance (like, you just have to watch those stupid bikes and hovercars maneuvering for five minutes until they finally get shot down so you can continue playing).

It's about core game loop first. When we have core game loop, people can play OpenApoc instead of OG and have full experience they would get from OG in OpenApoc.

And everything except the AI that belongs to the core game loop (funding, proper portal locations) just needs programming hours.

AI, however, is a different case.

We need to either reverse-engineer the OG AI code, to the point where we can understand how it actually works, or we need someone to study the OG AI and understand how it works. 

The result would be a document that outlines what do we want to create with the AI. What exactly should it do and how. Basically a programmer's design document, definition of what should be implemented.

And then we can get to implementing it. The framework should be there, I have coded a pretty long-stretching AI system that should allow to implement any additional logic into it. So then it would be implement, and optimize so it works quickly enough (like, it would most likely need to do a lot of tracing shots, where it calculates safe spots to hide behind or escape from, and that could get pretty taxing if not optimized).

Just need someone to really dedicate hours either into reverse-engineering or into playing with the OG under different scenarios and studying how does AI actually behave under different conditions and settings

**Comment by makus82** [comment]
I can join to this AI testing for sure ... 
But then if thinking if we think more broadly, the real problem is not even AI but programming hours and people who are needed for this... this is a task that we must solve....

**Comment by Istrebitel** [comment]
I would disagree here. Programming hours are free and available. Really! It's a matter of time - one year, two year, if this would be programming hours question we'd get it done. However, a programmer works best when there's a clear understanding what is there to program. With AI the problem is that there is just no such thing. You have to figure everything out from scratch first, and then implement. That's not programming and most programmers would not find this... like, attractive, so to speak? It's just a different kind of task and different field of expertise really. Notice how noone wants to implement this, they implement everything else but not this? That's why.

Personally for me, this is a chore, because it just feels like... you're wasting your time, because until you finally "get" it, you feel like you've spent hours and had no progress. That's demotivating. 

We need someone for whom this is a challenge he/she would like, to do it. Like, a person who likes to reverse engineer, or study how obscure games work, someone from speedrunning maybe, you know guys there spend ages just to find out how to bug out a game in any way possible - jumping at walls, doing weird inputs etc. Someone like that. If someone would do that, and we would have an understanding what to code - then it would just be a question of man hours put into it to get it done and we'd have a fully playable game - sans the crap most fans don't care about anyway.

**Comment by FilmBoy84** [comment]
On the subject of UnitAI; i'm still intermittently adding comments and video to your UnitAI thread Istrebitel, via the forums, documenting what user-side actions are performed by aliens, agents and human factions for both Final Release and Beta2

You can find the lists, comments, pics and even video of some of these things here
http://www.openapoc.org/threads/task-unit-behavior.185/

Still got to log more behaviours for other organisations, but agents and aliens are largely covered

**Comment by Istrebitel** [comment]
oh nice! that's what I'm talking about then. I will study it when I have time, but well done, that's EXACTLY what we need

**Comment by makus82** [comment]
> Personally for me, this is a chore, because it just feels like... you're wasting your time, because until you finally "get" it, you feel like you've spent hours and had no progress. That's demotivating.

That's what I'm talking about.
